### PR TITLE
[Merge-on-date: (TBA)] Add container log APIs to WebApps.json 

### PR DESCRIPTION
**Pull request by michimune**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json [hunk 1]
```diff
@@ -1913,6 +1913,84 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/containerlogs": {
+      "post": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Gets the last lines of docker logs for the given site",
```
Comment (fearthecowboy) [inline specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: did you want to add an appropriate `produces` element? (see [IANA Media Types](https://www.iana.org/assignments/media-types/media-types.xhtml) )

``` json
 "produces": [
     "text/plain"
  ],
```

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json [hunk 2]
```diff
@@ -1913,6 +1913,84 @@
...
+        "description": "Gets the last lines of docker logs for the given site",
+        "operationId": "WebApps_GetWebSiteContainerLogs",
```
Comment (fearthecowboy) [inline specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: And maybe here too?

``` json
 "produces": [
     "application/zip"
  ],
```

(and for the other APIs too...)

**Review by fearthecowboy — CHANGES_REQUESTED** [review]
Looks pretty good to me; you might want to add `produces` on the APIs

Also, new APIs require signoff from ARM. 

Adding @ravbhatnagar for his blessing.

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json [hunk 3]
```diff
@@ -1913,6 +1913,84 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "file"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/containerlogs/zip": {
```
Comment (ravbhatnagar) [inline specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: missing segments. containerLogs is the resource type as per the signature. So you need a resourceTypename segment on which you will call zip action. A better way to model this API would be return a pointer to storage blob where the zip can be downloaded from. We generally avoid content other than application/json flowing through ARM.

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json [hunk 4]
```diff
@@ -1913,6 +1913,84 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/containerlogs": {
```
Comment (ravbhatnagar) [inline specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: May be rename the action to getContainerLogs?

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json (hunk 1 above)
Comment (michimune) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Updated

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json (hunk 2 above)
Comment (michimune) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Updated

**Review by fearthecowboy — APPROVED** [review]

**Comment by fearthecowboy** [comment]
When @ravbhatnagar gives his ascent to your feedback, I can merge this.

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json (hunk 1 above)
Comment (jianghaolu) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Please let me know if I'm missing something here but this endpoint does seem to work for me:
```
POST https://management.azure.com/subscriptions/ffa52f27-be12-4cad-b1ea-c2c241b6cceb/resourceGroups/linux-test-appgroup/providers/Microsoft.Web/sites/linux-test-app?api-version=2016-08-01
405 Method Not Allowed 749.00 ms
{
"error": {
"code": "HttpMethodIsNotSupported",
"message": "The HTTP method 'POST' is not supported."
}
}
```

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json [hunk 5]
```diff
@@ -1913,6 +1913,90 @@
...
+            "description": "OK",
+            "schema": {
+              "type": "file"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/containerlogs/zip/download": {
+      "post": {
```
Comment (jianghaolu) [inline specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Also this one
```
POST https://management.azure.com/subscriptions/ffa52f27-be12-4cad-b1ea-c2c241b6cceb/resourceGroups/linux-test-appgroup/providers/Microsoft.Web/sites/linux-test-app/containerlogs/zip/download?api-version=2016-08-01
404 Not Found
The response was empty
```

**Comment by salameer** [comment]
@michimune and @ravbhatnagar 

What are the update on this  PR?

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json (hunk 1 above)
Comment (michimune) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: The deployment to production happened two weeks ago. I have confirmed it working with my testa pp. Can you please try again?

Comment (jianghaolu) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Seems like the endpoint is returning a base 64 encoded content - in this case, the return type shouldn't be `file` but `byte`. The `file` response type should be unencoded (gzipped okay). 

Also, the server is actually returning me `application/json` for content type.

File: specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json (hunk 2 above)
Comment (jianghaolu) [reply specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json]: Same for this one - get `application/json` for content type, which probably requires a server-side fix and the response should be `byte`.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Comment by michimune** [comment]
I can change the response schema type, but I have no idea how to specify the content type. I will need some investigation. Any advice would be helpful.

**Comment by jianghaolu** [comment]
@michimune It's a service issue - can you report to the service owner? The issue is these 2 APIs are sending non-JSON content with `application/json` content type header.

**Review by jianghaolu — APPROVED** [review]
The specs are approved. Waiting for the changes to be deployed to production.

**Comment by ravbhatnagar** [comment]
Looks good for ARM. Lets merge once the changes are deployed to prod.

**Review by ravbhatnagar — APPROVED** [review]

**Comment by fearthecowboy** [comment]
@michimune -- When is the expected date this will deployed to production?

**Comment by jianghaolu** [comment]
@fearthecowboy I just verified this is live now.

**Comment by fearthecowboy** [comment]
Thanks @jianghaolu ... merged!

**Pull request merged**